crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -346,11 +364,11 @@ fn impl_tuple_struct(
         TraitImpl::Implemented | TraitImpl::Custom(_) => reflect_attrs.get_partial_eq_impl(),
     };
 
-    let (impl_generics, ty_generics, _where_clause) = generics.split_for_impl();
+    let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
```

Comment:
For some reason, we weren't passing the where-clause to the impls. I don't know if that was intentional, but I made it so that it does (hopefully that's okay to do in this PR since it's small enough).

Reply:
Yep, this was a bug.